src/view-backend.c

```diff
@@ -109,6 +120,30 @@ wpe_view_backend_dispatch_frame_displayed(struct wpe_view_backend* backend)
         backend->backend_client->frame_displayed(backend->backend_client_data);
 }
 
+void
+wpe_view_backend_add_state(struct wpe_view_backend* backend, uint32_t state)
+{
+    uint32_t current_state = backend->state;
+    backend->state |= state;
+    if (current_state != backend->state)
```

Comment:
I think the intention would be clearer by calculating the new state first and assigning it to `backend->state` later when the newly calculated state is determined to be different. For me the following looks easier to understand:

```c
const uint32_t new_state = backend->state | state;
if (backend->state != new_state) {
    backend->state = new_state;
    wpe_view_backend_notify_state_changed(backend);
}
```

This would be just my preference, the current code is correct, though.